code/modules/recycling/conveyor2.dm

```diff
@@ -138,24 +136,36 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 	use_power(6)
 
 	//get the first 30 items in contents
-	affecting = list()
-	var/i = 0
-	var/list/items = loc.contents - src
-	for(var/item in items)
-		i++ // we're sure it's a real target to move at this point
-		if(i >= MAX_CONVEYOR_ITEMS_MOVE)
-			break
-		affecting.Add(item)
-
+	var/turf/locturf = loc
+	var/list/items = locturf.contents - src - locturf.lighting_object
```

Comment:
Is doing these list operations actually cheaper than just iterating it anyway.

Reply:
unlikely, this kind of list operations are expensive

Reply:
Current profiling averages:

on:
/obj/machinery/conveyor/process                                                        0.000        0.000        0.000        0.000        28866

off:
/obj/machinery/conveyor/process                                                        0.000        0.000        0.000        0.000         3502

on and filled:
/obj/machinery/conveyor/process                                                       0.000        0.000        0.000        0.000        12512


In addition it cuts the timers created when empty (in my small test example)